rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -781,6 +781,7 @@ def _predict(self, message: Message) -> Optional[Dict[Text, tf.Tensor]]:
 
         # create session data from message and convert it into a batch of 1
         model_data = self._create_model_data([message], training=False)
+        model_data.data = self._prepare_example_for_prediction(model_data)
```

Comment:
where is this comes from?

Reply:
this is required to make prediction work; otherwise we add a dictionary where the subdictionary value is empty (e.g., `"text": {"sequence" : some_sparse_matrix, "sentence": None, "length": []}`). This breaks the correspondence between shapes and surface form in `generator`.

Reply:
why did it work before?

Reply:
`rasa test nlu` didn't

Reply:
why do we add an empty value?

Reply:
we don't explicitly add it; but I'll double check, it might be in DIET's `_create_model_data`

Reply:
fixed it; it was a bug in `model_data.add_lengths`